src/shape/line.js

```diff
@@ -435,46 +435,24 @@ extend(ChartInternal.prototype, {
 
 		$$.mainCircle.exit().remove();
 
-		let createFn = $$.circle.create;
-
-		if ($$.hasValidPointType()) {
-			createFn = $$[$$.config.point_type].create;
-		} else if ($$.hasValidPointDrawMethods()) {
-			createFn = $$.config.point_type.create;
-		}
-
-		$$.mainCircle = createFn($$.mainCircle, $$.classCircle.bind($$), $$.pointR.bind($$), $$.color)
+		$$.mainCircle = $$.mainCircle.enter()
+			.append($$.point("create", this, $$.classCircle.bind($$), $$.pointR.bind($$), $$.color))
 			.merge($$.mainCircle)
 			.style("opacity", $$.initialOpacityForCircle.bind($$));
 	},
 
 	redrawCircle(cx, cy, withTransition, flow) {
 		const $$ = this;
 		const selectedCircles = $$.main.selectAll(`.${CLASS.selectedCircle}`);
-		const pointType = $$.config.point_type;
 
 		if (!$$.config.point_show) {
 			return [];
 		}
 
-		let updateFn = $$.circle.update;
+		const mainCircles = $$.mainCircle.each(
```

Comment:
The returned value should be a d3 transition object. Iterating all the `<circle>` elements just return a selection.
Because of this difference, `.flow()` not work as before.

If this can't be fixed, it can't be merged.

Reply:
@netil thanks for the detailed information.

`.flow()` should now be working with the latest changes.

Tell me what you think when you can.